### PR TITLE
Implement CountryFilter component for RankingsOverlay

### PR DESCRIPTION
**Pull request by EVAST9919**
Screenshot_1

### CODE REVIEW

File: osu.Game/Overlays/Rankings/CountryPill.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            ClearTransforms();
+            AutoSizeAxes = Axes.X;
+            this.FadeIn(duration, Easing.OutQuint);
+        }
+
+        public void Collapse()
+        {
+            ClearTransforms();
+            AutoSizeAxes = Axes.None;
+            this.ResizeWidthTo(0, duration, Easing.OutQuint);
```
Comment (bdach) [inline osu.Game/Overlays/Rankings/CountryPill.cs]: Is this necessary? It doesn't seem to be to me, and it ticks me off how this never gets restored in `Expand()`.

**Review by bdach — COMMENTED** [review]
Can't see too much fault with this; should be mostly fine.

File: osu.Game/Overlays/Rankings/CountryPill.cs (hunk 1 above)
Comment (EVAST9919) [reply osu.Game/Overlays/Rankings/CountryPill.cs]: It's restored by itself when setting `AutosizeAxes = Axes.X` including the fact that we have `AutosizeDuration` and `AutosizeEasing` in the ctor.
We already have some UI components with similar logic (`PlayerSettings` for example)

Comment (EVAST9919) [reply osu.Game/Overlays/Rankings/CountryPill.cs]: If by "necessary" you mean this Collapse/Expand animation then no, it's not. I was just thinking about what kind of animation we can apply here. Result seems pretty good to me, though.

Comment (bdach) [reply osu.Game/Overlays/Rankings/CountryPill.cs]: No, I just meant the resize call. Unless I screwed something up I didn't see any visual difference after removing it.

Comment (EVAST9919) [reply osu.Game/Overlays/Rankings/CountryPill.cs]: I just checked: there's a difference after removing it. May be it's not really noticeable.

**Comment by smoogipoo** [comment]
Required changes:
1. Using `BindableWithCurrent`.
2. Making things not inherit from `Container` or `CircularContainer`.

**Review by smoogipoo — APPROVED** [review]

**Review by bdach — APPROVED** [review]

**Pull request merged**